File: ruby/easy/excel_sheet_column_number.rb
# @param {String} s
# @return {Integer}
def title_to_number(s)
    res, fact = 0, 26
    s.each_char do |c|
        res += (c.ord - 'A'.ord + 1) * fact
        fact * 26
    end
    return res
end

File: javascript/easy/excelSheetColumnNumber.js
/**
 * @param {string} s
 * @return {number}
 */
var titleToNumber = function(s) {
    let result = 0, factor = 1;
    for(let i = s.length-1; i >= 0; i--) {
        result += ((s[i].charCodeAt() - 'A'.charCodeAt()) + 1) * factor;
        factor *= 26;
    }
    return result;
};
